Release/include/cpprest/details/basic_types.h

```diff
@@ -43,20 +48,358 @@ typedef uint64_t size64_t;
 typedef uint32_t HRESULT; // Needed for PPLX
 #endif
 
+#if defined _WIN32 && !defined _DLL
+// TEMPLATE CLASS allocator
+template<class _Ty>
+class allocator : public std::allocator<_Ty> {
+public:
+    typedef allocator<_Ty> _Alloc;
+    typedef std::allocator<_Ty> _Mybase;
+
+    template<class _Other>
+    struct rebind {
+        // convert this type to allocator<_Other>
+        typedef allocator<_Other> other;
+    };
+
+    allocator() CPPREST_NOEXCEPT : _Mybase() {
+        // construct default allocator (do nothing)
+    }
+
+    allocator(const _Alloc& _Right) CPPREST_NOEXCEPT : _Mybase(_Right) {
+        // construct by copying (do nothing)
+    }
+
+    allocator(_Alloc&& _Right) CPPREST_NOEXCEPT : _Mybase(std::move(_Right)) {
+        // construct by moving (do nothing)
+    }
+
+    template<class _Other>
+    allocator(const allocator<_Other>& _Right) CPPREST_NOEXCEPT : _Mybase(_Right) {
+        // construct from a related allocator (do nothing)
+    }
+
+    template<class _Other>
+    allocator(allocator<_Other>&& _Right) CPPREST_NOEXCEPT : _Mybase(std::forward<_Other>(_Right)) {
+        // construct from a related allocator (do nothing)
+    }
+
+    _Alloc& operator=(const _Alloc& _Right) {
+        // assign by copying
+        _Mybase::operator=(_Right);
+        return (*this);
+    }
+
+    _Alloc& operator=(_Alloc&& _Right) {
+        // assign by moving
+        _Mybase::operator=(std::move(_Right));
+        return (*this);
+    }
+
+    template<class _Other>
+    _Alloc& operator=(const allocator<_Other>&) {
+        // assign from a related allocator
+        _Mybase::operator=(std::forward<_Other>(_Right));
+        return (*this);
+    }
+
+    _DECLSPEC_ALLOCATOR pointer allocate(size_type _Count) {
```

Comment:
This is user code so it isn't allowed to be _Ugly.

Reply:
I don't understand, can you please explain how to resolve ?

Reply:
http://eel.is/c++draft/lex.name#3 says that _Ugly and __ugly names are reserved for use by the C++ implementation (compiler and standard library), not for user code. That said I think this is kind of moot given the compat problems here.